core/new-gui/src/app/dashboard/component/feature-container/saved-workflow-section/saved-workflow-section.component.ts

```diff
@@ -265,6 +290,39 @@ export class SavedWorkflowSectionComponent implements OnInit, OnChanges {
     this.router.navigate([`${ROUTER_WORKFLOW_CREATE_NEW_URL}`], { queryParams: { pid: this.pid } }).then(null);
   }
 
+  /**
+   * create a new workflow based on uploaded json file
+   */
+  public onClickUploadExistingWorkflowFromLocal = (file: NzUploadFile): boolean => {
+    var reader = new FileReader();
+    reader.readAsText(file as unknown as Blob);
+    reader.onload = () => {
+      try {
+        const result = reader.result;
+        if (typeof result !== "string") {
+          throw new Error("Incorrect format: file is not a string");
+        }
+        const workflowContent = JSON.parse(result).content as WorkflowContent;
+        const workflowName = JSON.parse(result).name as string;
```

Comment:
There are two names here, one is the `workflowName` in the JSON object, the other is the name of the json file. For the navigation bar upload in #1582, since the contents do not have `workflowName`, the file name will be adopted as the workflow name. The behavior here is different from #1582 in that the name will always be the content in the json file, even if the user changes the file name to a different one, and this may cause confusions. @zuozhiw do we want to keep this or change it to be consistent with #1582?

Reply:
Please follow the comment from @Xiao-zhen-Liu and unify the format of this PR to be consistent with #1582